executor/memtable_reader.go

```diff
@@ -293,6 +293,17 @@ type clusterServerInfoRetriever struct {
 
 // retrieve implements the memTableRetriever interface
 func (e *clusterServerInfoRetriever) retrieve(ctx context.Context, sctx sessionctx.Context) ([][]types.Datum, error) {
+	switch e.serverInfoType {
+	case diagnosticspb.ServerInfoType_LoadInfo,
+		diagnosticspb.ServerInfoType_SystemInfo:
+		if !hasPriv(sctx, mysql.ProcessPriv) {
+			return nil, plannercore.ErrSpecificAccessDenied.GenWithStackByArgs("Process")
```

Comment:
All privilege names reported in `ErrSpecificAccessDenied` everywhere else are in uppercase, can you keep them the same?

Reply:
Unfortunately I am just continuing the inconsistent naming :(
```
tidb> create user u1;
tidb> grant process,config on *.* to u1;
tidb> show grants for u1;
+-----------------------------------------+
| Grants for u1@%                         |
+-----------------------------------------+
| GRANT Process,CONFIG ON *.* TO 'u1'@'%' |
+-----------------------------------------+
1 row in set (0,00 sec)
```
But I could change it to uppercase, but then it would not be consistent with the grants output instead. What do you prefer?

Reply:
I think uppercase. The name being lower here is actually a bug which makes porting tests harder - I've looked at fixing it, but there is a bit of work involved.

Reply:
Ah, now I see the issue more clear:
```
tidb> show grants for u1;
+-----------------------------------------+
| Grants for u1@%                         |
+-----------------------------------------+
| GRANT Process,CONFIG ON *.* TO 'u1'@'%' |
+-----------------------------------------+
1 row in set (0,00 sec)
```
vs
```
mysql> show grants for u1;
+-----------------------------------------+
| Grants for u1@%                         |
+-----------------------------------------+
| GRANT PROCESS, SUPER ON *.* TO `u1`@`%` |
+-----------------------------------------+
1 row in set (0,00 sec)
```

But the confusion of mine comes from:
```
mysql> show privileges;
+----------------------------+---------------------------------------+-------------------------------------------------------+
| Privilege                  | Context                               | Comment                                               |
+----------------------------+---------------------------------------+-------------------------------------------------------+
...
| Grant option               | Databases,Tables,Functions,Procedures | To give to other users those privileges you possess   |
| Index                      | Tables                                | To create or drop indexes                             |
| Process                    | Server Admin                          | To view the plain text of currently executing queries |
| Super                      | Server Admin                          | To use KILL thread, SET GLOBAL, CHANGE MASTER, etc.   |
| Usage                      | Server Admin                          | No privileges - allow connect only                    |
...
| SESSION_VARIABLES_ADMIN    | Server Admin                          |                                                       |
| CLONE_ADMIN                | Server Admin                          |                                                       |
| FLUSH_STATUS               | Server Admin                          |                                                       |
+----------------------------+---------------------------------------+-------------------------------------------------------+
62 rows in set (0,00 sec)
```
and
```
tidb> show privileges;
+-----------------------------+---------------------------------------+-------------------------------------------------------+
| Privilege                   | Context                               | Comment                                               |
+-----------------------------+---------------------------------------+-------------------------------------------------------+
| Alter                       | Tables                                | To alter the table                                    |
...
| Process                     | Server Admin                          | To view the plain text of currently executing queries |
...
| Super                       | Server Admin                          | To use KILL thread, SET GLOBAL, CHANGE MASTER, etc.   |
...
| RESTRICTED_TABLES_ADMIN     | Server Admin                          |                                                       |
...
+-----------------------------+---------------------------------------+-------------------------------------------------------+
42 rows in set (0,00 sec)
```

I will create an issue for correcting the 'privileges in grants should all be in UPPER case' (while keeping the 'show privileges' in mixed case depending on privilege.)

Reply:
Reported as https://github.com/pingcap/tidb/issues/26275

Reply:
I updated the patch to use the upper case `PROCESS`